src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs

```diff
@@ -332,26 +332,42 @@ internal ProgressTracker(string activityName, Func<ResourceManagerRestRestClient
                 this.ProgressRecord = new ProgressRecord(activityId: 0, activity: activityName, statusDescription: "Starting - 0.00% completed.");
             }
 
+            internal void SetProgressPercentageAndWriteProgress(double percentage)
+            {
+                this.SetProgressRecordPercentComplete(percentage);
+                this.WriteProgressAction(this.ProgressRecord);
+            }
+
             /// <summary>
             /// Logs the fact that the operation has progressed.
             /// </summary>
             /// <param name="result">The operation result</param>
-            internal void UpdateProgress(TrackingOperationResult result)
+            internal void UpdateProgress(TrackingOperationResult result, bool isResourceCreateOrUpdate)
```

Comment:
I would change isResourceCreateOrUpdate to something more generic, like isLongRunning. This is because we need the progress for not just create or update, but for other operations like resource move